### PR TITLE
Test cleanups and fixes

### PR DESCRIPTION
**Pull request by jleclanche**
This PR achieves the following things:
- Drops support for Django 1.9 (1.8 is still supported)
- Cleans up some Django <1.8 hacks
- Adds per-Django-version pypi classifiers (each classifier advertises a supported Django major version)
- Upgrades the Django 1.11 test version, now that it's been fixed
- Sets up testing using Tox on Travis. This means the test matrix is handled by tox instead of Travis, and we just tell travis which environments to pair the matrix with. This pattern makes it possible to run the matrix locally or on other CIs easily.
- Creates new tox environments `qa` and `docs`, for QA checks and docs building, respectively. Instead of `make qa` / `make docs`, the command is now `tox -e qa` / `tox -e docs`. This is also separated on Travis so that docs/qa failures don't fail *all* the builds, but only their respective one (easier to understand what's wrong).
- Adds testing on django master (currently 2.0), with failures allowed. This gives us a heads up on what to expect in the future, without making the builds red. The `fast_finish` travis keyword tells travis not to wait for the djangomaster build to report results.
- ~~Switches to yarn instead of npm to install standardjs. This speeds up the build considerably.~~ *apparently this isn't possible on non-js travis envs...*

Please don't squash :)

### CODE REVIEW

**Comment by jleclanche** [comment]
@pennersr thoughts?

**Comment by lorddaedra** [comment]
@jleclanche  LGTM

but I suggest to save 3.3 support for tests with Django 1.8

File: tox.ini [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+	py27: mock >= 1.0.1
+	django18: Django>=1.8,<1.9
+	django110: Django>=1.10,<1.11
+	django111: Django>=1.11,<2.0
+	djangomaster: https://github.com/django/django/archive/master.tar.gz
 commands =
-    coverage run manage.py test allauth
-    coverage report
-    coverage html
+	coverage run manage.py test allauth
```
Comment (lorddaedra) [inline tox.ini]: Can we add "-Wall" here to show Deprecation Warnings?..

Comment (jleclanche) [inline tox.ini]: Can you pass -Wall to coverage? I think it's just for python

Comment (lorddaedra) [inline tox.ini]: You are right.

Then let's add 

```
[testenv]
setenv =
    PYTHONWARNINGS = all
```

**Comment by jleclanche** [comment]
I did a bad push on my master and now I can't reopen the PR. I'll file it again.

**Pull request closed without merging**